File: Python/isValid.py
class Solution:
    def isValid(self, s: str) -> bool:
        s = list(s)
        stack = []

        if len(s) < 2: 
            return False

        for _s in s:
            try:
                if _s != ")" and _s != "}" and _s != "]":
                    stack.append(_s)
                else:
                    if _s == ")" and stack[-1] != "(": return False
                    elif _s == "}" and stack[-1] != "{": return False
                    elif _s == "]" and stack[-1] != "[": return False
                    else: stack.pop(-1)
            except: return False

        if len(stack) > 0: return False
        return True
        

File: Java/isValid.java
import java.util.Stack;

class Solution {
    public boolean isValid(String s) {
        Stack<Character> holder = new Stack<Character> ();

        if (s.length() > 1) {
            for (int i=0; i<s.length(); i++) {
                boolean flag = false;
                if (holder.size() > 0) {
                    if (s.charAt(i) == ']') {
                        if (holder.peek() == '[') {
                            flag = true;
                        }
                    } 
                    else if (s.charAt(i) == '}'){
                        if (holder.peek() == '{') {
                            flag = true;
                        }
                    }
                    else if (s.charAt(i) == ')') {
                        if (holder.peek() == '(') {
                            flag = true;
                        } 
                    }
                }
                
                if (flag == false) {
                    holder.push(s.charAt(i));
                } else { holder.pop(); }
                
            }
        } else {return false; }

        if (holder.size() > 0) {
            return false;
        } else { return true; }
    }
}

class Main {
    public static void main(String[] args) {
        String s = "(])";
        Solution solution = new Solution();
        System.out.print(solution.isValid(s));
    }
}